test/GoodGhostingPolygonTest.js

```diff
@@ -0,0 +1,708 @@
+        });
+
+        it("reverts if funds were already redeemed", async () => {
+            await joinGamePaySegmentsAndComplete(player1, whitelistedPlayerConfig[0][player1].index, whitelistedPlayerConfig[0][player1].proof);
+            await goodGhosting.redeemFromExternalPool({ from: player1 });
+            await truffleAssert.reverts(goodGhosting.redeemFromExternalPool({ from: player1 }), "Redeem operation already happened for the game");
+        });
+
+        it("allows to redeem from external pool when game is completed", async () => {
+            await joinGamePaySegmentsAndComplete(player1, whitelistedPlayerConfig[0][player1].index, whitelistedPlayerConfig[0][player1].proof);
+            truffleAssert.passes(goodGhosting.redeemFromExternalPool, "Couldn't redeem from external pool");
+        });
+
+        it("transfer funds to contract then redeems from external pool", async () => {
+            const expectedBalance = web3.utils.toBN(segmentPayment * segmentCount);
+            await joinGamePaySegmentsAndComplete(player1, whitelistedPlayerConfig[0][player1].index, whitelistedPlayerConfig[0][player1].proof);
+            await goodGhosting.redeemFromExternalPool({ from: player2 });
+            const contractsDaiBalance = await token.balanceOf(goodGhosting.address);
+            // No interest is generated during tests so far, so contract balance must equals the amount deposited.
+            assert(expectedBalance.eq(contractsDaiBalance));
+        });
+
+        it("emits event FundsRedeemedFromExternalPool when redeem is successful", async () => {
+            await joinGamePaySegmentsAndComplete(player1, whitelistedPlayerConfig[0][player1].index, whitelistedPlayerConfig[0][player1].proof);
+            const result = await goodGhosting.redeemFromExternalPool({ from: player1 });
+            const contractsDaiBalance = await token.balanceOf(goodGhosting.address);
+            truffleAssert.eventEmitted(
+                result,
+                "FundsRedeemedFromExternalPool",
+                (ev) => new BN(ev.totalAmount).eq(new BN(contractsDaiBalance)),
+                "FundsRedeemedFromExternalPool event should be emitted when funds are redeemed from external pool",
+            );
+        });
+
+        it("emits WinnersAnnouncement event when redeem is successful", async () => { // having test with only 1 player for now
+            await joinGamePaySegmentsAndComplete(player1, whitelistedPlayerConfig[0][player1].index, whitelistedPlayerConfig[0][player1].proof);
+            const result = await goodGhosting.redeemFromExternalPool({ from: player1 });
+            truffleAssert.eventEmitted(result, "WinnersAnnouncement", (ev) => {
+                return ev.winners[0] === player1;
+            }, "WinnersAnnouncement event should be emitted when funds are redeemed from external pool");
+        });
```

Comment:
we should probably add a test case that confirmed rewards were claimed as expected.